### PR TITLE
#174797519 Login with Google and Facebook

### PR DESCRIPTION
**Pull request by jpirumvaa**
### What does this PR do?
Allow users to login with Google and Facebook

### Description of Task to be completed?

1. Pull the repo to get the project's latest version
2. Create a branch to work on
3. Configure Project on Google APIs
4. Write PassportJS codes to allow google authentication
5. Provide token after google login
6. Write PassportJS codes to allow Facebook authentication
7. Provide token after Facebook login
8. Add, commit, and push changes to GitHub

### How should this be manually tested?
Clone a repo, run `npm run dev` to run the project. After having a project running on a port using `localhost:PORTNO` to run the project in the browser. Go to `http://localhost:PORTNO/users/auth/google` if you want to sign in with Google or `http://localhost:3000/users/auth/facebook` if you want to sign in with Facebook. Choose the account and continue.
After successfully allowing the app to use your data, you will receive the user's information and a token.
### What are the relevant pivotal tracker stories?
[#174797519](https://www.pivotaltracker.com/n/projects/2465784/stories/174797519)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -33,11 +36,15 @@ if (!isProduction) {
...
 
+app.use('/', googleRouter);
+app.use('/', facebookRoute);
+
 app.get('/', (req, res, next) => {
   res.status(200).send({ message: res.__('Welcome to barefoot nomad') });
 });
 app.use('/api', routes);
 
+
```
Comment (kipropbrian) [inline src/index.js]: Please remove unnecessary new lines

File: src/middlewares/facebookSign.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import FacebookStrategy from 'passport-facebook';
+import passport from 'passport';
+import { User } from '../database/models';
+
+passport.use(new FacebookStrategy({
+  clientID: process.env.FB_APP_ID,
+  clientSecret: process.env.FB_CLIENT_SECRETE,
+  callbackURL: 'http://localhost:3000/users/auth/facebook/callback',
```
Comment (kipropbrian) [inline src/middlewares/facebookSign.js]: What is this????

Comment (kipropbrian) [reply src/middlewares/facebookSign.js]: Make sure that such URL's are variables, stored in .env

File: .env-example [hunk 3]
```diff
@@ -17,3 +17,8 @@ DB_PROD_USERNAME="user"
 DB_PROD_PASSWORD="password"
 DB_PROD_NAME="database"
 DB_PROD_HOST="localhost"
+
+GOOGLE_CLIENTID= 923132586967-kapi48i3amtl8ves6ak5ffk2vt5dlm9d.apps.googleusercontent.com
+GOOGLE_CLIENT_SECRETE= GG9XR4aezYJMeD5Zz5bKu8w2
+FB_APP_ID= 735172470394706
+FB_CLIENT_SECRETE= ba64da6ab1a42444dd021af01b29cfb3
```
Comment (kipropbrian) [inline .env-example]: Do not expose secrets in the .env files, kindly use dummy data and provide actual secrets on slack

**Comment by dushimeemma** [comment]
Fix hound issues and work on tests

**Review by Habinezajanvier — APPROVED** [review]
ok

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "editor.formatOnSave": true
+}
```
Comment (valens10) [inline .vscode/settings.json]: Add newline

Comment (kipropbrian) [reply .vscode/settings.json]: @jpirumvaa see above

**Pull request merged**